src/Psalm/Codebase.php

```diff
@@ -1159,4 +1160,31 @@ public function isTypeContainedByType(
     ): bool {
         return TypeAnalyzer::isContainedBy($this, $input_type, $container_type);
     }
+
+    /**
+     * @psalm-suppress PossiblyUnusedMethod
+     */
+    public function canTypeBeContainedByType(
+        Type\Union $input_type,
+        Type\Union $container_type
+    ): bool {
+        return TypeAnalyzer::canBeContainedBy($this, $input_type, $container_type);
+    }
+
+    /**
+     * @return array{Type\Union,Type\Union}
+     * @psalm-suppress PossiblyUnusedMethod
+     */
+    public function getKeyValueParamsForTraversableObject(Type\Atomic $type): array
```

Comment:
Would you mind add a test or two for this? That’d save needing a PossiblyUnusedMethod annotation.

Reply:
Would you suggest a specific test file?

Reply:
I'd actually encourage you to create a new file, `CodebaseTest` for these specific things

Reply:
Added. I'm getting errors trying to check any classes though, so I resorted to testing with built-in types only.

Reply:
Errors like 'cannot get classstorage for exception'